dbus_next/_private/unmarshaller.py

```diff
@@ -66,6 +69,24 @@ def align(self, n):
             return b''
         return self.read(padding)
 
+    def read_endian(self, _=None):
+        unix_fds = array.array("i")
+
+        if self.sock:
+            try:
```

Comment:
This code path might run multiple times for a single unmarshalling of a message. That will happen in the case there was a blocking error. The previously read data will be read from a buffer cache instead.

One solution would be to cache the fds on the unmarshaller instance and then skip reading the fds from the socket in the case that the fd cache is present.

Reply:
I added that cache to unmarshaller, please have a look if this is what you had in mind.